core/src/main/java/org/apache/carbondata/core/dictionary/server/DictionaryServer.java

```diff
@@ -146,21 +147,31 @@ public void shutdown() throws Exception {
    * Write dictionary to the store.
    * @throws Exception
    */
-  public void writeDictionary() throws Exception {
+  public void writeDictionary(CarbonTable carbonTable) throws Exception {
     DictionaryMessage key = new DictionaryMessage();
     key.setType(DictionaryMessageType.WRITE_DICTIONARY);
-    dictionaryServerHandler.processMessage(key);
+    dictionaryServerHandler.processMessage(key, carbonTable);
+  }
+
+  public void initializeDictionaryGenerator(CarbonTable carbonTable) throws Exception {
+    DictionaryMessage key = new DictionaryMessage();
+    key.setType(DictionaryMessageType.TABLE_INTIALIZATION);
+    key.setColumnName("dummy");
+    key.setTableUniqueId(carbonTable.getCarbonTableIdentifier().getTableId());
+    key.setTableUniqueName(carbonTable.getTableUniqueName());
```

Comment:
check if tableUniqueName is required...I think we should be able to work using tableUniqueId